pet_clinic_billing_service/billing_service/views.py

```diff
@@ -154,6 +156,42 @@ def log(self, data):
             # Don't raise the exception to avoid disrupting the main flow
 
 
+class SummaryViewSet(viewsets.ViewSet):
+    def list(self, request, pk=None):
+        span = trace.get_current_span()
+        
+        # Always set request counter to 1
+        span.set_attribute("billing_summary_request", 1)
+
+         # Set num_summaries based on current minute
+        current_minute = timezone.now().minute
+        num_summaries = 50 if current_minute % 5 == 0 else 2
+        
+        # Use random cache key to simulate high cache miss rate
+        cache_key = f'billing_summary_last_7_days_{random.randint(1, num_summaries)}'
+        summary = cache.get(cache_key)
+        
+        if summary is None:
+            # Cache miss
+            span.set_attribute("billing_summary_cache_hit", 0)
+            
+            time.sleep(2)
```

Comment:
What does the sleep do here? In the API operation, we usually want to keep the latency low.